### PR TITLE
rsx_debugger: replace malloc with buffer cache

### PR DESCRIPTION
**Pull request by Megamouse**
- Use buffer cache instead of allocating a raw buffer
- This also happens to fix a memory leak in the stencil buffer widget

### CODE REVIEW

File: rpcs3/rpcs3qt/rsx_debugger.cpp [hunk 1]
```diff
@@ -577,7 +573,8 @@ void rsx_debugger::OnClickDrawCalls()
 		if (width && height && !draw_call.depth_stencil[1].empty())
 		{
 			const std::span<const std::byte> orig_buffer = draw_call.depth_stencil[1];
-			u8* buffer = static_cast<u8*>(std::malloc(4ULL * width * height));
```
Comment (kd-11) [inline rpcs3/rpcs3qt/rsx_debugger.cpp:580]: How'd this ever get merged? :eyes:

Comment (Megamouse) [reply rpcs3/rpcs3qt/rsx_debugger.cpp:580]: probably as part of the first Qt or wx pr xD

Comment (elad335) [reply rpcs3/rpcs3qt/rsx_debugger.cpp:580]: The whole function is dead code I'm pretty sure.

**Comment by Megamouse** [comment]
I hope this works, because i didn't test

**Pull request merged**